include/Core/Posts/Metabox/DeletePostsByCategoryMetabox.php

```diff
@@ -0,0 +1,165 @@
+<?php
+namespace BulkWP\BulkDelete\Core\Posts\Metabox;
+
+use BulkWP\BulkDelete\Core\Posts\PostsMetabox;
+
+defined( 'ABSPATH' ) || exit; // Exit if accessed directly.
+
+/**
+ * Delete Posts by Category Metabox.
+ *
+ * @since 6.0.0
+ */
+class DeletePostsByCategoryMetabox extends PostsMetabox {
+	/**
+	 * @var int Limit for the categories.
+	 */
+	private $cat_limit = 50;
+	protected function initialize() {
+		$this->item_type     = 'posts';
+		$this->field_slug    = 'cats';
+		$this->meta_box_slug = 'bd_by_category';
+		$this->action        = 'delete_posts_by_category';
+		$this->cron_hook     = 'do-bulk-delete-cat';
+		$this->scheduler_url = 'http://bulkwp.com/addons/scheduler-for-deleting-posts-by-category/?utm_source=wpadmin&utm_campaign=BulkDelete&utm_medium=buynow&utm_content=bd-sc';
+		$this->messages      = array(
+			'box_label' => __( 'By Post Category', 'bulk-delete' ),
+			'scheduled' => __( 'The selected posts are scheduled for deletion', 'bulk-delete' ),
+		);
+	}
+
+	/**
+	 * Render Delete posts by category box.
+	 */
+	public function render() {
+        ?>
+        <!-- Category Start-->
+        <h4><?php _e( 'Select the post type from which you want to delete posts by category', 'bulk-delete' ); ?></h4>
+        <fieldset class="options">
+            <table class="optiontable">
+				<?php bd_render_post_type_dropdown( 'cats' ); ?>
```

Comment:
Create a helper method in `PostMetabox` class with name `render_post_type_dropdown`. In side that method call the `bd_render_post_type_dropdown` method.

That way this new method can be used in 'Delete Posts by Tag' and other metaboxes.